pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -413,6 +433,22 @@ public static SelectionResults renderSelectionResultsWithoutOrdering(List<Serial
     return new SelectionResults(selectionColumns, rows);
   }
 
+  /**
+   * Render the selection rows to a formatted {@link ResultTable} object for selection queries without
+   * <code>ORDER BY</code>. (Broker side)
+   * <p>{@link ResultTable} object will be used to build the broker response.
+   * <p>Should be called after method "reduceWithoutOrdering()".
+   *
+   * @param rows unformatted selection rows.
+   * @param dataSchema data schema.
+   * @return {@link ResultTable} object results.
+   */
+  public static ResultTable renderResultTableWithoutOrdering(List<Serializable[]> rows, DataSchema dataSchema) {
+    List<Object[]> resultRows = new ArrayList<>(rows.size());
```

Comment:
I think we need to add a major TODO to avoid this copy.

Reply:
A TODO has been added in SelectionDataTableReducer in 2 places, to work on changing Selection code from Serializable[] to Object[]